axum/src/routing/path_router.rs

```diff
@@ -36,6 +36,29 @@ where
     }
 }
 
+// Validates a path in compile time, used with the vpath macro.
+// Allow dead code needed because this is only used within the vpath macro
+// and if there's no call to it, then, there's no users of it.
+#[allow(dead_code)]
+const fn validate_static_path(path: &'static str) -> &'static str {
+    if path.is_empty() {
+        panic!("Paths must start with a `/`. Use \"/\" for root routes")
+    }
+    if path.as_bytes()[0] != b'/' {
+        panic!("Paths must start with /");
+    }
+    path
+}
+
+#[macro_export]
+/// usage: .route(vpath("/")) // compiles fine
+/// .route(vpath("porato")) // compilation error -> "Paths must start with a /"
+macro_rules! vpath {
+    ($e:expr) => {
+        const { validate_static_path($e) }
```

Comment:
This won’t work, because it will look for a function called `validate_static_path` in the caller’s scope, and not in axum’s scope. What you would need to do is make `validate_static_path` `pub` and also `#[doc(hidden)]`, and then refer to it as `$crate::path::to::validate_static_path`.

To follow convention, you might want to re-export this function at the crate root as `__private_validate_static_path`.

Reply:
thanks, I'll do that.